src/Rings/mpoly-graded.jl

```diff
@@ -1595,15 +1755,37 @@ function dehomogenization(F::MPolyElem_dec, R::MPolyRing, pos::Int)
   return finish(B)
 end
 
+
+
+function _dehomogenization(F::MPolyElem_dec, R::MPolyRing, pos::Int, m::Int)
+  B = MPolyBuildCtx(R)
+  for (c,e) = zip(coefficients(F), exponent_vectors(F))
+      for i = (pos+m-1):-1:pos
+         deleteat!(e, i)
+      end
```

Comment:
You can remove multiple elements at once (and this can be more efficient):
```suggestion
       deleteat!(e, pos:pos+m-1)
```